src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -192,6 +192,35 @@ public void openFolder() {
         }
     }
 
+    public boolean performNameConflictCheck() {
+        /*
+        Get the filepath for current linkedFile and truncate it to only have the filename and
+        not the filepath. Compare to new suggested filename and return boolean result.
+         */
+        Path file = Paths.get(this.linkedFile.getLink());
+        String currentFileName = file.getFileName().toString();
+        String suggestedFileName = this.linkedFileHandler.getSuggestedFileName();
+
+        return currentFileName.equals(suggestedFileName);
+    }
+
+    public boolean performDirectoryConflictCheck() {
+        /*
+        Get the suggested filepath for current linkedFile and compare it to the existing filepath.
+        Return boolean result.
+         */
+        //Get the new path for the file
+        Optional<Path> newDir = databaseContext.getFirstExistingFileDir(filePreferences);
+        Path newDirectory = newDir.get();
```

Comment:
The `get` method throws an exception if the optional is empty. I would propose to rewrite the code as `return OptionalUtil.equals(newDir, currentDirectory, Files::isSameFile);` and add the following method to OptionalUtil
````java
public <T,U> boolean equals(Optional<T> left, Optional<U> right, BiPredicate<T,U> equality) {
   if(!left.isPresent()) {
       return !right.isPresent();
   } else {
       if (right.isPresent()) {
           return equality.test(left.get(), right.get());
      } else {
           return false;
      }
   }
}
````
(there might be syntax errors in the above code)